#27901 breaks mikroe_* shields overlay

The change in order #27901 will put the cart before the horse when building with overlays.

**To Reproduce**

> west build -b lpcxpresso55s69_cpu0 samples/net/sockets/echo_server --pristine -- -DSHIELD=mikroe_eth_click

**Expected behavior**
Should build.

**Impact**
Does not build.

> Error: lpcxpresso55s69_cpu0.dts.pre.tmp:447.15-25 Label or path eth_click not found
> FATAL ERROR: Syntax error parsing input tree
> CMake Error at /mnt/c/1.Code/zephyrproject/zephyr/cmake/dts.cmake:211 (message):
>   command failed with return code: 1

Excerpt from temporary dts file (lpcxpresso55s69_cpu0.dts.pre.tmp):
```
/delete-node/ &eth_click; <---- board's overlay
&mikrobus_spi {
 status = "okay";
 eth_click: eth_click@1 {
  compatible = "microchip,enc28j60";
  reg = <0x1>;
  local-mac-address = [00 00 00 01 02 03];
  spi-max-frequency = <10000000>;
  int-gpios = <&mikrobus_header 7 (1 << 0)>;
  label = "ETH_CLICK";
 };
};
&mikrobus_spi { <---- shield's overlay
 status = "okay";
 eth_click: eth_click@0 {
  compatible = "microchip,enc28j60";
  reg = <0x0>;
  local-mac-address = [00 00 00 01 02 03];
  spi-max-frequency = <10000000>;
  int-gpios = <&mikrobus_header 7 (1 << 0)>;
  label = "ETH_CLICK";
 };
};
```

**Possible resolution**
Change the overlay include order or change the board overlay

--- a/cmake/app/boilerplate.cmake
+++ b/cmake/app/boilerplate.cmake
@@ -380,6 +380,17 @@ foreach(root ${BOARD_ROOT})
       list(GET shields_refs_list ${_idx} s_path)
       get_filename_component(s_dir ${s_path} DIRECTORY)
 
+      # if shield config flag is on, add shield overlay to the shield overlays
+      # list and dts_fixup file to the shield fixup file
+      list(APPEND
+        shield_dts_files
+        ${shield_dir}/${s_path}
+        )
+      list(APPEND
+        shield_dts_fixups
+        ${shield_dir}/${s_dir}/dts_fixup.h
+        )
+
       # search for shield/boards/board.overlay file
       if(EXISTS ${shield_dir}/${s_dir}/boards/${BOARD}.overlay)
         # add shield/board overlay to the shield overlays list
@@ -397,17 +408,6 @@ foreach(root ${BOARD_ROOT})
           ${shield_dir}/${s_dir}/boards/${s}/${BOARD}.overlay
           )
       endif()
-
-      # if shield config flag is on, add shield overlay to the shield overlays
-      # list and dts_fixup file to the shield fixup file
-      list(APPEND
-        shield_dts_files
-        ${shield_dir}/${s_path}
-        )
-      list(APPEND
-        shield_dts_fixups
-        ${shield_dir}/${s_dir}/dts_fixup.h
-        )
 
       # search for shield/shield.conf file
       if(EXISTS ${shield_dir}/${s_dir}/${s}.conf)

